static/app/utils/formatters.spec.tsx

```diff
@@ -202,6 +202,16 @@ describe('formatAbbreviatedNumber()', function () {
     expect(formatAbbreviatedNumber(1500)).toBe('1.5k');
     expect(formatAbbreviatedNumber(1213122)).toBe('1.2m');
   });
+
+  it('should round to set amount of significant digits', () => {
+    expect(formatAbbreviatedNumber(100.12, 3)).toBe('100');
+    expect(formatAbbreviatedNumber(1500, 3)).toBe('1.5k');
```

Comment:
This happens because of the `parseFloat` call, which is done so we remove exponential notation that can be generated by `number.toPrecision` (for ex. `1.230e+5`)